dev/NumberBox/NumberBox.cpp

```diff
@@ -0,0 +1,555 @@
+{
+StepValue(false);
+}
+
+void NumberBox::OnSpinUpClick(winrt::IInspectable const& sender, winrt::RoutedEventArgs const& args)
+{
+    StepValue(true);
+}
+
+void NumberBox::OnNumberBoxKeyUp(winrt::IInspectable const& sender, winrt::KeyRoutedEventArgs const& args)
+{
+    switch (args.Key()) {
+    case winrt::VirtualKey::Up:
+    case winrt::VirtualKey::GamepadDPadUp:
+        StepValue(true);
+        break;
+    case winrt::VirtualKey::Down:
+    case winrt::VirtualKey::GamepadDPadDown:
+        StepValue(false);
+        break;
+    }
+}
+
+void NumberBox::OnScroll(winrt::IInspectable const& sender, winrt::PointerRoutedEventArgs const& args) {
+    if (HyperScrollEnabled()) {
+        int delta = args.GetCurrentPoint(*this).Properties().MouseWheelDelta();
+        if (delta > 0)
+        {
+            StepValue(true);
+        }
+        else if (delta < 0)
+        {
+            StepValue(false);
+        }
+    }
+}
+
+// Increments or decrements value by StepFrequency, wrapping if necessary
+void NumberBox::StepValue(bool isPositive)
+{
+    // Validating input before and after
+    ValidateInput();
+    double oldVal = Value();
+    double newVal = oldVal;
+
+    if (isPositive)
+    {
+        newVal += StepFrequency();
+    }
+    else
+    {
+        newVal -= StepFrequency();
+    }
+
+    // MinMaxMode Wrapping
+    if (MinMaxMode() == winrt::NumberBoxMinMaxMode::WrapEnabled && GetBoundState(newVal) != BoundState::InBounds)
+    {
+        while (newVal > MaxValue())
+        {
+            newVal = MinValue() + (newVal - MaxValue()) - 1;
+        }
+        while (newVal < MinValue())
+        {
+            newVal = MaxValue() - abs(newVal - MinValue()) + 1;
+        }
+    }
+    // Input Overwriting - Coerce to min or max
+    else if (BasicValidationMode() == winrt::NumberBoxBasicValidationMode::InvalidInputOverwritten && GetBoundState(newVal) != BoundState::InBounds)
+    {
+        if (newVal > MaxValue() && (MinMaxMode() == winrt::NumberBoxMinMaxMode::MaxEnabled || MinMaxMode() == winrt::NumberBoxMinMaxMode::MinAndMaxEnabled))
+        {
+            newVal = MaxValue();
+        }
+        else if (newVal < MinValue() && (MinMaxMode() == winrt::NumberBoxMinMaxMode::MinEnabled || MinMaxMode() == winrt::NumberBoxMinMaxMode::MinAndMaxEnabled))
+        {
+            newVal = MinValue();
+        }
+    }
+    
+    // Safeguard for floating point imprecision errors
+    int StepFreqSigDigits = ComputePrecisionRounderSigDigits(newVal);
+    m_stepPrecisionRounder.SignificantDigits(StepFreqSigDigits);
+    newVal = m_stepPrecisionRounder.RoundDouble(newVal);
+
+    // Update Text and Revalidate new value
+    Value(newVal);
+    UpdateTextToValue();
+    ValidateInput();
+}
+
+
+// Check if text resembles formulaic input to determine if parser should be executed
+bool NumberBox::IsFormulaic(const winrt::hstring& in)
+{
+    std::wstring input(in);
+    std::wregex formula(L"^([0-9()\\s]*[+-/*^%]+[0-9()\\s]*)+$");
```

Comment:
This should be more permissive. As in, if there's anything resembling an operator or parenthesis then send it to parsing -- it doesn't need to regex check for numbers, etc since that would mean we're checking in two places.